In [67]:
import pandas as pd



In [5]:
#df.head()     # первые 5 строк
# df.tail()     # последние 5 строк
# df.shape      # (строки, столбцы)
# df.columns    # имена столбцов
# df.info()     # информация о типах данных

df = pd.read_csv("Titanic Dataset.csv")
df.shape

(1309, 14)

"Titanic Dataset.csv" qatorlar soni 1309 ta va ustunlar soni 14 ekan

In [66]:
all_people = df["name"]
alive_people = df[df["survived"] == 1]
alive_female = df[(df["sex"] == "female") & (df["survived"] == 1)]["name"]
alive_male = df[(df["sex"] == "male") & (df["survived"] == 1)]["name"]
print(len(all_people)) # Jami odamlar soni 1309 ta
print(len(alive_people)) # Trik qolgan odamlar soni 500 ta
print(len(alive_female)) # Trik qolgan ayollar soni 339 ta
print(len(alive_male)) # Trik qolgan ayollar soni 161 ta


1309
500
339
161


Jami odamlar soni 1309 ta ekan. 
Ulardan faqat 500 tasi trik qolgan. 
339 tasi ayollar va 161 tasi erkaklarni tashkil etadi.

In [61]:
alive_pclass_1 = df[(df["pclass"] == 1) & (df["survived"] == 1)]["name"]
alive_pclass_2 = df[(df["pclass"] == 2) & (df["survived"] == 1)]["name"]
alive_pclass_3 = df[(df["pclass"] == 3) & (df["survived"] == 1)]["name"]
print(len(alive_pclass_1))  # 1 - portdan kelgan trik odamlar soni 200 ta
print(len(alive_pclass_2))  # 2 - portdan kelgan trik odamlar soni 119 ta
print(len(alive_pclass_3))  # 3 - portdan kelgan trik odamlar soni 181 ta 


200
119
181


Trik qolgan odamlarning 
200 tasi 1-portdan kelgan,
119 tasi 2-portdan kelgan,
181 tasi 3-portdan kelgan.

In [ ]:
number_people_less_19 = df[(df["age"] <= 19)]["name"]
alive_younger_children = df[(df["age"] <= 12) & (df["survived"] == 1)]["name"]
alive_teenage = df[(df["age"] >= 13) & (df["age"] <= 19) & (df["survived"] == 1)]["name"]
print(len(number_people_less_19)) # 19 yoshdan kichik odamlar soni 225 ta
print(len(alive_younger_children)) # Trik 12 yoshdan kichik yosh bollar soni 54 ta ekan.
print(len(alive_teenage)) # Trik 13 yoshdan 19 yoshgacha bo'lgan odamlar soni 52 ta


225
54
52


19 yoshdan kichik odamlar soni 225 ta. 
Ulardan 106 tasi trik qolgan 
Shu trik qolgan odamlarning 54 tasi 12 yoshdan kichik odamlar hisoblanadi.
Qolgan 52 tasi esa 13 yosh va 19 yosh o'rtasidagi odamlar hisoblanadi.

In [ ]:
df["ticket"] = pd.to_numeric(df["ticket"], errors="coerce")

avg = df["ticket"].mean()
alive_avg_ticket_people = df[(df["ticket"] > avg) & (df["survived"] == 1)]["name"]
alive_avg_ticket_male = df[(df["ticket"] > avg) & (df["survived"] == 1) & (df["sex"] == "male")]["name"]

print(f"Ticketning o'rtacha narxi: {int(avg)}")
print(len(alive_avg_ticket_people))  # o'rtacha tickets narx sotib olingan tirik odamlar soni 100 ta ekan.
print(len(alive_avg_ticket_male))  # o'rtacha tickets narx sotib olingan tirik erkaklar soni 35 ta ekan.




Ticketning o'rtacha narxi: 249039
100
35


Ticketning o'rtacha narxi: 249039
Ticketlarning o'rtacha narxdan baland sotib olingan tirik odamlar soni 100 ta ekan.
Ulardan 35 tasi erkak ekan
Ulardan 65 tasi ayol ekan